backend/app/assets/javascripts/spree/backend/global/animate_css.es6

```diff
@@ -0,0 +1,19 @@
+/* eslint-disable no-unused-vars */
+
+//
+// Handle clearing out of animation styles from complete animation.
+const animateCSS = (element, animation, speed, prefix = 'animate__') =>
+  new Promise((resolve) => {
+    const animationName = `${prefix}${animation}`
+    const node = document.querySelector(element)
+
+    node.classList.add(`${prefix}animated`, animationName, prefix + speed)
+
+    function handleAnimationEnd(event) {
+      event.stopPropagation()
```

Comment:
Do you need to use `stopPropagation()` here? It's always tricky to stop propagation, because, it kind-of breaks the expected functionality of browsers: you can attach a listener on `body` and you'll have the listener called for every event. This could be used, for example, for analytics.

Reply:
@vfonic This is just the stock boiler plate javascript from the Animate.css site for handling the animations classes, when I get change I'll cast my eye over this again in greater detail.

Reply:
Ah I see!

Perhaps you could just add a comment that links to where this was taken from:
https://animate.style/#javascript

In that case perhaps it's better we leave it as-is. I'd just add in the same comment a bit about how to use it:


```js
// Taken from: https://animate.style/#javascript
// Use like this:
//   animateCSS('.my-element', 'bounce');
// or:
//   animateCSS('.my-element', 'bounce').then((message) => {
//      // Do something after the animation
//  })
const animateCSS = (element, animation, prefix = 'animate__') =>
  // ...
```